src/editor/components/modals/revision-list.jsx

```diff
@@ -40,17 +41,30 @@ class RevisionListUnconnected extends React.Component {
   }
 
   render() {
+    if (this.props.revisionList !== undefined) {
+      if (
+        this.props.hideLocalChanges &&
+        this.props.selectedRevisionId === undefined
+      ) {
+        if (this.props.revisionList.length > 0) {
+          this.revisionClicked(this.props.revisionList[0].id);
+        }
+      }
+    }
```

Comment:
I don't think this should be necessary? We seemed able to show the right set of changes without firing the event on render before, not sure why that should be different now?

Reply:
In case this does not take place, it just shows "No difference". With this, modal defaults to last revision, showing its difference with previous version.

Reply:
Ah, I see. I think we might need to put some of this calculation at the reducer level, which will let the sub-components (like this one and the actual revision viewer) act accordingly. You probably want to put the logic that tries to figure out whether we have local changes here:

https://github.com/iodide-project/iodide/blob/52170a96a5bce10828cf113e395cebdf4365bc8b/src/editor/reducers/notebook-reducer.js#L81

You may also want to perform some more intelligent logic to pick a sensible default value for selectedRevisionId (should be `undefined` if we have local-only changes, some integer value if we don't).

In general, we try to avoid business logic inside a render() function, so if you find yourself needing to do that it's a good sign that things are a bit off-track.

Hope that wasn't too confusing. Please feel free to ask more questions if anything else is unclear!

Reply:
Yes, that will make the code more consistent. I will implement that.